bika/lims/browser/bika_listing.py

```diff
@@ -1592,6 +1595,48 @@ def hide_hidden_attributes(self):
                 new_states.append(state)
         self.bika_listing.review_states = new_states
 
+    def get_top_right_hooks(self):
+        """
+        Get adapters (hooks) that implements ITopRightListingHook.
+        The information got from those adapters will be placed right-over the
+        list.
+
+        :return: html code
+        """
+        adapters = getAdapters((self, ), ITopRightListingHook)
+        export_options = ''
+        for name, adapter in adapters:
+            export_options += adapter(self.request)
+        return export_options
+
+    def get_top_left_hooks(self):
+        """
+        Get adapters (hooks) that implements ITopLeftListingHook.
+        The information got from those adapters will be placed left-over the
+        list.
+
+        :return: html code
+        """
+        adapters = getAdapters((self, ), ITopLeftListingHook)
+        export_options = ''
+        for name, adapter in adapters:
+            export_options += adapter(self.request)
+        return export_options
+
+    def get_top_wide_hooks(self):
+        """
+        Get adapters (hooks) that implements ITopWideListingHook.
+        The information got from those adapters will be placed wide-over the
+        list.
+
+        :return: html code
+        """
+        adapters = getAdapters((self, ), ITopWideListingHook)
+        export_options = ''
+        for name, adapter in adapters:
+            export_options += adapter(self.request)
+        return export_options
+
```

Comment:
Since you are repeating this logic several times, and with the aim to reduce duplicate code, I suggest to add this function:

```python
def get_adapters_html(adapter_provider=None):
    if not adapter_provider:
        return ''
    html = list()
    adapters = getAdapters(self, ), adapter_provider)
    for name, adapter in adapters:
        adapter_html = adapter(self.request)
        html.append(adapter_html)
    return ''.join(html)
```

Then, logic from inside `get_top_wide_hooks`, `get_top_left_hooks` and `get_top_right_hooks` could be simplified as follows:

```python
def get_top_left_hooks(self):
    return self.get_adapters_html(ITopLeftListingHook)
```

Note that adapters may do other things than exporting content, so variable names from inside the functions should be generic (do not use `export_options`).